contracts/rust/src/jf_helpers.rs

```diff
@@ -0,0 +1,581 @@
+//! Helpers for connecting types between Jellyfish and Solidity.
+//! Usually used during differential testing (via FFI).
+
+use anyhow::Result;
+use ark_bn254::{Bn254, Fq, Fr, G1Affine, G2Affine};
+use ark_ff::{Fp2, MontFp, PrimeField};
+use ark_std::{rand::Rng, str::FromStr, UniformRand};
+pub use diff_test_bn254::{field_to_u256, u256_to_field, ParsedG1Point};
+use ethers::{
+    abi::AbiDecode,
+    prelude::{AbiError, EthAbiCodec, EthAbiType},
+    types::{Bytes, H256, U256},
+};
+use itertools::izip;
+use jf_plonk::proof_system::structs::{OpenKey, Proof, ProofEvaluations, VerifyingKey};
+use jf_plonk::proof_system::{PlonkKzgSnark, UniversalSNARK};
+use jf_plonk::testing_apis::Challenges;
+use jf_plonk::{constants::KECCAK256_STATE_SIZE, transcript::SolidityTranscript};
+use jf_primitives::pcs::prelude::{Commitment, UnivariateUniversalParams};
+use jf_relation::{Arithmetization, Circuit, PlonkCircuit};
+use num_bigint::BigUint;
+use num_traits::Num;
+
+// constant in hex string copied from hardcoded constants from solidity contracts
+
+const COSET: [&str; 5] = [
+    "1",
+    "2f8dd1f1a7583c42c4e12a44e110404c73ca6c94813f85835da4fb7bb1301d4a",
+    "1ee678a0470a75a6eaa8fe837060498ba828a3703b311d0f77f010424afeb025",
+    "2042a587a90c187b0a087c03e29c968b950b1db26d5c82d666905a6895790c0a",
+    "2e2b91456103698adf57b799969dea1c8f739da5d8d40dd3eb9222db7c81e881",
+];
+
+// H: G2Affine(x: Fp2, y:Fp2), x = x0 + u * x1, y = y0 + u * y1
+// NOTE: extra careful with discrepancy with current version of BN254.G2Point
```

Comment:
This is not true anymore right?

Reply:
true, removed in [6171539](https://github.com/EspressoSystems/espresso-sequencer/pull/854/commits/61715390c68c49dcf05f32a2505fc782957d9628)